File: program128.cpp
#include<iostream>
using namespace std;

class Number
{
    private :
        int iNo;

    public :

        void Accept()
        {
            cout << "Enter the value" << endl;
            cin >> this->iNo;
        }

        void Display()
        {
            cout << "value is: " << this->iNo << endl;
        }

        int Factor()
        {
            int iFact = 1;
            int iCnt = 0;
            for(iCnt = 1; iCnt <= iNo; iCnt++)
            {
                iFact *= iCnt;
            }
            return iFact;
        }

};

int main()
{
    Number nobj1;
    Number nobj2;

    int iRet = 0;

    nobj1.Accept();
    nobj1.Display();
    iRet = nobj1.Factor();
    cout << "Factorial is : " << iRet << endl;

    nobj2.Accept();
    nobj2.Display();
    iRet = nobj2.Factor();
    cout << "Factorial is : " << iRet << endl;

    return 0;
}

File: program128.java
// Calculate Factorial

import java.lang.*;
import java.util.*;

class Number
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the Number : ");

        this.iNo = sobj.nextInt();
    }
    
    public void Display()
    {
        System.out.println("The value is :" + this.iNo);
    }

    public boolean CheckPerfect()
    {
        int iCnt = 0;
        int iSum = 0;

        for(iCnt = 1; iCnt <= ((this.iNo) / 2) && (iSum <= this.iNo); iCnt++)
        {
            if(this.iNo % iCnt == 0)
            {
                iSum += iCnt;
            }
        }

        if(iSum == this.iNo)
        {
            return true;
        }
        else
        {
            return false;
        }

    }

}

class program128
{
    public static void main(String arg[])
    {
        boolean bRet = false;

        Number obj = new Number();

        obj.Accept();
        obj.Display();
        bRet = obj.CheckPerfect();

        if(bRet == true)
        {
            System.out.println("It is a Perfect Number");
        }
        else
        {
            System.out.println("It is not a Perfect Number");
        }

    }

}